### PR TITLE
[FAB-17605] Unhandled InitFactories error

### PR DESCRIPTION
**Pull request by Eggsyz**
#### Type of change

- Bug fix

#### Description

A project using func InitFactories, it output errors when build it. But many places do not handle errors.
#### Related issues

[FAB-17605](https://jira.hyperledger.org/browse/FAB-17605?jql=project%20%3D%20FAB)

### CODE REVIEW

File: internal/peer/chaincode/common_test.go [hunk 1]
```diff
@@ -184,8 +188,11 @@ func TestGetOrdererEndpointFail(t *testing.T) {
 	assert.NoError(t, err)
 
 	mockchain := "mockchain"
-	factory.InitFactories(nil)
-
+	// Init the BCCSP
+	err = factory.InitFactories(nil)
+	if err != nil {
+		assert.Error(t, err, "Could not initialize BCCSP Factories")
```
Comment (denyeart) [inline internal/peer/chaincode/common_test.go]: Error is not expected here, so why assert it?
BTW expected error would go before actual error in assert.Error().

File: internal/peer/chaincode/common_test.go [hunk 2]
```diff
@@ -159,7 +159,11 @@ func TestGetOrdererEndpointFromConfigTx(t *testing.T) {
 	assert.NoError(t, err)
 
 	mockchain := "mockchain"
-	factory.InitFactories(nil)
+	// Init the BCCSP
+	err = factory.InitFactories(nil)
+	if err != nil {
+		assert.Error(t, err, "Could not initialize BCCSP Factories")
```
Comment (denyeart) [inline internal/peer/chaincode/common_test.go]: See other comment.

File: cmd/configtxgen/main_test.go [hunk 3]
```diff
@@ -177,12 +177,16 @@ func TestBlockFlags(t *testing.T) {
 }
 
 func TestPrintOrg(t *testing.T) {
-	factory.InitFactories(nil)
+	// Init the BCCSP
+	err := factory.InitFactories(nil)
+	if err != nil {
+		assert.Error(t, err, "Could not initialize BCCSP Factories")
```
Comment (denyeart) [inline cmd/configtxgen/main_test.go]: See other comment.

File: cmd/configtxgen/main.go [hunk 4]
```diff
@@ -255,7 +255,11 @@ func main() {
 	}()
 
 	logger.Info("Loading configuration")
-	factory.InitFactories(nil)
+	// Init the BCCSP
+	err := factory.InitFactories(nil)
+	if err != nil {
+		logger.Fatalf("Could not initialize BCCSP Factories [%s]", err)
```
Comment (denyeart) [inline cmd/configtxgen/main.go]: Why does this one log it, but the other ones panic?

**Review by denyeart — CHANGES_REQUESTED** [review]

File: internal/peer/chaincode/common_test.go (hunk 1 above)
Comment (lindluni) [reply internal/peer/chaincode/common_test.go]: Use assert.NoError(t, err) which will flag unexpected introduced behavior

File: cmd/configtxgen/main_test.go (hunk 3 above)
Comment (Eggsyz) [reply cmd/configtxgen/main_test.go]: thank you. I will fix it

**Pull request closed without merging**